src/backsolve.jl

```diff
@@ -138,22 +193,27 @@ function CuSparseSM(
     n_bytes = outL[]::UInt64
     buffer = CUDA.zeros(UInt8, n_bytes)
 
+    # Uncomment when we will have a routine cusparseSpSM_updateMatrix
+    # buffer_L = CUDA.zeros(UInt8, n_bytes)
+    # buffer_U = CUDA.zeros(UInt8, n_bytes)
+    # CUSPARSE.cusparseSpSM_analysis(
+    #     CUSPARSE.handle(), transa, transx, Ref{T}(alpha), descL, descX, descX, T, algo, spsm_L, buffer_L,
+    # )
+    # CUSPARSE.cusparseSpSM_analysis(
+    #     CUSPARSE.handle(), transa, transx, Ref{T}(alpha), descU, descX, descX, T, algo, spsm_U, buffer_U,
+    # )
+
     return CuSparseSM(algo, transa, descL, descU, spsm_L, spsm_U, buffer)
 end
 
 function backsolve!(s::CuSparseSM, A::CUSPARSE.CuSparseMatrixCSR{T}, X::CuMatrix{T}) where T
     m,n = A.dims
-    alpha = T(1.0)
+    alpha = one(T)
 
     descX = CUSPARSE.CuDenseMatrixDescriptor(X)
     transx = 'N'
 
-    operations = if s.transa == 'N'
-        [(s.descL, s.infoL), (s.descU, s.infoU)]
-    elseif s.transa == 'T'
-        [(s.descU, s.infoU), (s.descL, s.infoL)]
-    end
-
+    operations = s.transa == 'N' ? [(s.descL, s.infoL), (s.descU, s.infoU)] : [(s.descU, s.infoU), (s.descL, s.infoL)]
```

Comment:
I guess the interface remains the same as before for `CuSparseSM`. How about using `cusparseSpSM_solve` in that function, and updating the analysis apart in the interface `RFLU`?  
https://github.com/exanauts/CUSOLVERRF.jl/blob/master/src/interface.jl#L85

Reply:
Do you call multiple time `SpSM` without updating the `rf` structure`?
In that case we can reuse the buffer but otherwise we will gain nothing to move the computation of the `SpSM` analyis.

Reply:
Yes, we call `SpSM` multiple times inside a for loop (on large instances we cannot instantiate a buffer large enough ).